private/bufpkg/bufplugin/bufpluginconfig/bufpluginconfig.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bufpluginconfig defines the buf.plugin.yaml file.
+package bufpluginconfig
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bufbuild/buf/private/bufpkg/bufplugin/bufpluginref"
+	"github.com/bufbuild/buf/private/pkg/encoding"
+	"github.com/bufbuild/buf/private/pkg/storage"
+)
+
+const (
+	// ExternalConfigFilePath is the default configuration file path for v1.
+	ExternalConfigFilePath = "buf.plugin.yaml"
+	// V1Version is the version string used to indicate the v1 version of the buf.plugin.yaml file.
+	V1Version = "v1"
+)
+
+var (
+	// AllConfigFilePaths are all acceptable config file paths without overrides.
+	//
+	// These are in the order we should check.
+	AllConfigFilePaths = []string{
+		ExternalConfigFilePath,
+	}
+)
+
+// Config is the plugin config.
+type Config struct {
+	// Name is the name of the plugin (e.g. 'buf.build/protocolbuffers/go').
+	Name bufpluginref.PluginIdentity
+	// Options is the set of options available to the plugin.
+	//
+	// For now, all options are string values. This could eventually
+	// support other types (like JSON Schema and Terraform variables),
+	// where strings are the default value unless otherwise specified.
+	//
+	// Note that some legacy plugins don't always express their options
+	// as key value pairs. For example, protoc-gen-java has an option
+	// that can be passed like so:
+	//
+	//  java_opt=annotate_code
+	//
+	// In those cases, the option value in this map will be set to
```

Comment:
Questionable, but I understand it

Reply:
Agreed on it being questionable - left a note in the summary that mentions we might change this behavior later. Feels fine to start like this for now, and we can iterate on it as we go.